memfs/memory.go

```diff
@@ -270,6 +270,13 @@ func (f *file) Close() error {
 	return nil
 }
 
+func (f *file) Truncate(size int64) error {
+	if size < int64(len(f.content.bytes)) {
+		f.content.bytes = f.content.bytes[:size]
```

Comment:
Truncate is supposed to extend the file if `size` is bigger than the current size, right?  At least that's what the equivalent syscall does.

Reply:
Fixed and added a test.